controllers/secretsync_controller_test.go

```diff
@@ -91,7 +91,298 @@ func TestReconcile(t *testing.T) {
+							{
+								SourcePath: "foo",
+								TargetKey:  "bar",
+							},
+						},
+					},
+				},
+			},
+			secret: &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "sse2esecret",
+					Namespace: "default",
+				},
+				Data: map[string][]byte{
+					"foo": []byte("bar"),
+				},
+			},
+			expectedError:       true,
+			expectedErrorString: "secretproviderclasses.secrets-store.csi.x-k8s.io \"test-spc\" not found",
+		},
+		{
+			name: "failed to get provider client",
+			secretProviderClassToProcess: &secretsstorecsiv1.SecretProviderClass{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-spc",
+					Namespace: "default",
+				},
+				Spec: secretsstorecsiv1.SecretProviderClassSpec{
+					Provider: "invalid-fake-provider",
+					Parameters: map[string]string{
+						"foo": "v1",
+					},
+				},
+			},
+			secretSyncToProcess: &secretsyncv1alpha1.SecretSync{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "sse2esecret",
+					Namespace: "default",
+				},
+				Spec: secretsyncv1alpha1.SecretSyncSpec{
+					ServiceAccountName:      "default",
+					SecretProviderClassName: "test-spc",
+					SecretObject: secretsyncv1alpha1.SecretObject{
+						Type: "Opaque",
+						Data: []secretsyncv1alpha1.SecretObjectData{
+							{
+								SourcePath: "foo",
+								TargetKey:  "bar",
+							},
+						},
+					},
+				},
+			},
+			secret: &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "sse2esecret",
+					Namespace: "default",
+				},
+				Data: map[string][]byte{
+					"foo": []byte("bar"),
+				},
+			},
+			expectedError:       true,
+			expectedErrorString: "provider not found: provider \"invalid-fake-provider\"",
+		},
+		{
+			name: "invalid SecretObjectData returns validation error",
+			secretProviderClassToProcess: &secretsstorecsiv1.SecretProviderClass{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-spc",
+					Namespace: "default",
+				},
+				Spec: secretsstorecsiv1.SecretProviderClassSpec{
+					Provider: "fake-provider",
+					Parameters: map[string]string{
+						"foo": "v1",
+					},
+				},
+			},
+			secretSyncToProcess: &secretsyncv1alpha1.SecretSync{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "sse2esecret",
+					Namespace: "default",
+				},
+				Spec: secretsyncv1alpha1.SecretSyncSpec{
+					ServiceAccountName:      "default",
+					SecretProviderClassName: "test-spc",
+					SecretObject: secretsyncv1alpha1.SecretObject{
+						Type: "Opaque",
+						Data: []secretsyncv1alpha1.SecretObjectData{
+							{
+								SourcePath: "foo",
+								TargetKey:  "",
+							},
+						},
+					},
+				},
+			},
+			secret: &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "sse2esecret",
+					Namespace: "default",
+				},
+				Data: map[string][]byte{
+					"foo": []byte("bar"),
+				},
+			},
+			expectedError:       true,
+			expectedErrorString: "key in secretObject.data is empty",
 		},
```

Comment:
Need unit tests for

https://github.com/kubernetes-sigs/secrets-store-sync-controller/blob/569742b2ec0ff963a19ccc94b93c209197c03970/controllers/secretsync_controller.go#L242-L245

Reply:
https://github.com/kubernetes-sigs/secrets-store-sync-controller/issues/44

Reply:
I added test in this PR itself